dali/operators/reader/loader/video/frames_decoder_test.cc

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) 2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <exception>
+
+#include "dali/test/dali_test_config.h"
+#include "dali/core/error_handling.h"
+#include "dali/operators/reader/loader/video/video_test_base.h"
+#include "dali/operators/reader/loader/video/frames_decoder.h"
+
+
+namespace dali {
+class FramesDecoderTest : public VideoTestBase {
+};
+
+
+TEST_F(FramesDecoderTest, ConstantFrameRate) {
+    std::string path = testing::dali_extra_path() + "/db/video/cfr/test_1.mp4";
+
+    // Create file, build index
+    FramesDecoder file(path);
+
+    ASSERT_EQ(file.Height(), 720);
+    ASSERT_EQ(file.Width(), 1280);
+    ASSERT_EQ(file.Channels(), 3);
+    ASSERT_EQ(file.NumFrames(), 50);
+
+    std::vector<uint8_t> frame(file.FrameSize());
+
+    // Read first frame
+    file.ReadNextFrame(frame.data());
+    this->ComapreFrames(frame.data(), this->GetCfrFrame(0, 0), file.FrameSize());
+
+    // Seek to frame
+    file.SeekFrame(25);
+    file.ReadNextFrame(frame.data());
+    this->ComapreFrames(frame.data(), this->GetCfrFrame(0, 25), file.FrameSize());
+
+    // Seek back to frame
+    file.SeekFrame(12);
+    file.ReadNextFrame(frame.data());
+    this->ComapreFrames(frame.data(), this->GetCfrFrame(0, 12), file.FrameSize());
+
+    // Seek to last frame (flush frame)
+    file.SeekFrame(49);
+    file.ReadNextFrame(frame.data());
+    this->ComapreFrames(frame.data(), this->GetCfrFrame(0, 49), file.FrameSize());
+
+    // Wrap around to first frame
+    ASSERT_FALSE(file.ReadNextFrame(frame.data()));
```

Comment:
I thought it's been removed?

Reply:
It was removed from the class, but it stays in the test. It is just called with manual `Reset`. As you can see, this checks that the value returned is false, when you go outside of the file.

Reply:
I see the manual reset below, but the comment looks as if it applied to this line, which even checks that we _can't_ wrap around.

Reply:
There is a comment at the top of four line block that does what the comment says.